### PR TITLE
Enable dependabot for updating the `github.com/gardener/gardener` dependency

### PR DESCRIPTION
**Pull request by ialidzhikov**
/kind enhancement

**What this PR does / why we need it**:
I noticed that currently the gvisor integration tests against K8s 1.27 Shoots fails with
```

2023-08-04T05:47:35.471598675Z stdout F   In [BeforeEach] at: /src/vendor/github.com/gardener/gardener/test/framework/shootframework.go:96 @ 08/04/23 05:47:34.658
2023-08-04T05:47:35.471592966Z stdout F   occurred
2023-08-04T05:47:35.471587868Z stdout F       }
2023-08-04T05:47:35.471582416Z stdout F           },
2023-08-04T05:47:35.47157768Z stdout F               },
2023-08-04T05:47:35.471572758Z stdout F                   },
2023-08-04T05:47:35.471566592Z stdout F                       s: "unsupported kubernetes version \"v1.27.4\"",
2023-08-04T05:47:35.471560488Z stdout F                   err: <*errors.errorString | 0xc000125830>{
2023-08-04T05:47:35.471555398Z stdout F                   msg: "error discovering kubernetes version: unsupported kubernetes version \"v1.27.4\"",
2023-08-04T05:47:35.471548559Z stdout F               err: <*fmt.wrapError | 0xc000b566e0>{
2023-08-04T05:47:35.471542265Z stdout F               msg: "could not construct Shoot client: error discovering kubernetes version: unsupported kubernetes version \"v1.27.4\"",
2023-08-04T05:47:35.471536218Z stdout F           err: <*fmt.wrapError | 0xc000b56720>{
2023-08-04T05:47:35.471530755Z stdout F           ctxError: <context.deadlineExceededError>{},
2023-08-04T05:47:35.47152382Z stdout F       {
2023-08-04T05:47:35.471517461Z stdout F       retry failed with context deadline exceeded, last error: could not construct Shoot client: error discovering kubernetes version: unsupported kubernetes version "v1.27.4"
2023-08-04T05:47:35.471506925Z stdout F       <*retry.Error | 0xc000d16760>: 
```

The issue is described in https://github.com/gardener/gardener-extension-runtime-gvisor/issues/27. 

TL;DR short term, gardener/gardener >= 1.71.0 has to be vendored (to include https://github.com/gardener/gardener/commit/7489574cf91eb6c5ad5f2ed3597104b7d0163145).

I decided to enable dependabot for this repo to automate as much as possible the process of vendoring the `github.com/gardener/gardener` dependency.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/6328

**Special notes for your reviewer**:
N/A

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Review by MrBatschner — APPROVED** [review]
Thank you.
/lgtm
